File: 1108-defanging-an-ip-address/1108-defanging-an-ip-address.js
/**
 * @param {string} address
 * @return {string}
 */
var defangIPaddr = (address) => {
   let result = '';
   for (let i = 0; i < address.length; i++) {
     if (address.charAt(i) === '.') {
        result += "[.]";
        continue;
     }
     result += address.charAt(i);
   }
   return result; 
};

File: 1108-defanging-an-ip-address/1108-defanging-an-ip-address.c
char * defangIPaddr(char * address){
    int len = strlen(address), j = 0;
    char* newStr = malloc(sizeof(char) * (len + 7));
    for(int i = 0; i < len; i++) {
        if (address[i] == '.') {
            newStr[j++] = '[';
            newStr[j++] = '.';
            newStr[j++] = ']';
        } else {
            newStr[j++] = address[i];
        }
    }
    newStr[j] = '\0';
    return newStr;
}